### DISCUSSION TITLE
How can i open preview in new tab

### DISCUSSION THREAD

**Discussion by ajitkaur8b** [vueform/builder#93] [Questions] [answered]
I want to open preview in new tab as well as want to save the code in db for forms that i have created. Also i need to give unique id to every form that i have created.

**Comment by adamberecz** [comment, marked as answer]
You can access the export object of the builder from `localStorage` under `vueform-builder` key. It contains `schema` and `form` properties that you can [use](https://builder.vueform.com/docs/rendering#rendering) in a new window to preview a form. You need to add an external button somewhere outside the builder to open the new window.

Saving can be achieved by subscribing to the `@save` event of `<VueformBuilder>` component as described [here](https://builder.vueform.com/docs/saving). You can add a unique key to each form before saving. Later, you can [load](https://builder.vueform.com/docs/loading) back the forms to the builder or used the saved JSON to [render](https://builder.vueform.com/docs/rendering) forms.
